bin/deployables

```diff
@@ -570,27 +570,64 @@ function copy_lambda_to_s3 () {
 
     aws_configure
 
-    aws --profile "${DEPLOY_AWS_PROFILE}" \
-        s3 cp \
-        ${DEPLOY_LAMBDA_ZIP_FULLPATH} \
-        s3://${DEPLOY_LAMBDA_S3_BUCKET}/${DEPLOY_LAMBDA_FUNCTION_NAME}/archive/${DEPLOY_LAMBDA_FUNCTION_NAME}-${DEPLOY_SHA1}.zip
-
-    if [[ "$?" != "0" ]] ; then
-        echo "Error: Failed to copy lambda zip to S3 (${DEPLOY_SHA1})"
-        exit 1
-    fi
+    # aws --profile "${DEPLOY_AWS_PROFILE}" \
+    #     s3 cp \
+    #     ${DEPLOY_LAMBDA_ZIP_FULLPATH} \
+    #     s3://${DEPLOY_LAMBDA_S3_BUCKET}/${DEPLOY_LAMBDA_FUNCTION_NAME}/archive/${DEPLOY_LAMBDA_FUNCTION_NAME}-${DEPLOY_SHA1}.zip
+    #
+    # if [[ "$?" != "0" ]] ; then
+    #     echo "Error: Failed to copy lambda zip to S3 (${DEPLOY_SHA1})"
+    #     exit 1
+    # fi
 
-    if [ ! -z "${DEPLOY_LAMBDA_ZIP_VERSION}" ] ; then
+  #  if [ ! -z "${DEPLOY_LAMBDA_ZIP_VERSION}" ] ; then
       aws --profile "${DEPLOY_AWS_PROFILE}" \
           s3 cp \
           ${DEPLOY_LAMBDA_ZIP_FULLPATH} \
-          s3://${DEPLOY_LAMBDA_S3_BUCKET}/${DEPLOY_LAMBDA_FUNCTION_NAME}/${DEPLOY_LAMBDA_FUNCTION_NAME}-${DEPLOY_LAMBDA_ZIP_VERSION}.zip
+          s3://${DEPLOY_LAMBDA_S3_BUCKET}/${DEPLOY_LAMBDA_FUNCTION_NAME}/${DEPLOY_LAMBDA_FUNCTION_NAME}.zip
 
       if [[ "$?" != "0" ]] ; then
-          echo "Error: Failed to copy lambda zip to S3 (${DEPLOY_LAMBDA_ZIP_VERSION})"
+          echo "Error: Failed to copy lambda zip to S3 (${DEPLOY_LAMBDA_ZIP_FULLPATH})"
           exit 1
       fi
 
+  #  fi
+
+}
+
+function deploy_lambda_s3 () {
+    # check environment variables
+    if [ -z "${DEPLOY_AWS_ACCOUNT}" ] ; then
+        echo "Error: DEPLOY_AWS_ACCOUNT is required"
+        exit 1
+    fi
+
+    if [ -z "${DEPLOY_LAMBDA_FUNCTION_NAME}" ] ; then
+        echo "Error: DEPLOY_LAMBDA_FUNCTION_NAME is required"
+        exit 1
+    fi
+
+    if [ -z "${DEPLOY_LAMBDA_S3_BUCKET}" ] ; then
+        echo "Error: DEPLOY_LAMBDA_S3_BUCKET is required"
+        exit 1
+    fi
+
+    # setup aws
+    aws_configure
+
+    # update lambda function
+    lambda_function_arn=$( \
+        aws lambda update-function-code \
+            --profile "${DEPLOY_AWS_PROFILE}" \
+            --function-name "${DEPLOY_LAMBDA_FUNCTION_NAME}" \
+            --s3-bucket "${DEPLOY_LAMBDA_S3_BUCKET}" \
+            --s3-key "${DEPLOY_LAMBDA_FUNCTION_NAME}/${DEPLOY_LAMBDA_FUNCTION_NAME}.zip" \
+            
+    )
```

Comment:
aws-cli is configured by deployables to output json.  you might need to put a `| $JQ foo` in here to check the return value.

Reply:
I think I just copy pasta'ed that from the other deploy code further down in the file, I guess I just missed the $JQ part.

Reply:
👍